request.go

```diff
@@ -308,13 +309,16 @@ func (req *Request) multipartPayload(fieldname, filename string, data io.Reader)
 
 // isErrBreaking returns false if the given error is involved with an option called by the user
 func (req *Request) isErrBreaking(err error) bool {
-	if req.retryOnEOFError && err == io.EOF {
+	switch {
+	case strings.Contains(err.Error(), "read: connection reset by peer"),
+		req.retryOnEOFError && err == io.EOF:
 		return false
+	default:
+		return true
 	}
-	return true
 }
 
-// WithReaderPayload sets the given payload for the Request
+// WithReaderPayloads sets the given payload for the Request
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported function WithReaderPayload should be of the form "WithReaderPayload ..."